docs/concepts/datatypes_datashapes.qmd

```diff
@@ -0,0 +1,202 @@
+# Datatypes and Datashapes
+
+Every value in Ibis has two important properties: a type and shape.
+
+The type is probably familiar to you. It is something like
+
+- `Integer`
+- `Floating`
+- `String`
+- `Array`
+
+The shape is one of
+
+- `Scalar` (a single value)
+- `Column` (a series of values)
+
+## Datatype Flavors
+
+For some datatypes, there are further options that define them.
+For instance, `Integer` values can be signed or unsigned, and
+they have a precision. For example, "uint8", "int64", etc.
+These flavors don't affect their capabilities
+(eg both signed and unsigned ints have a `.abs()` method),
+but the flavor does impact how the underlying backend performs the computation.
+
+## Capabilities
+
+Depending on the combination of datatype and datashape, a value has
+different capabilities. For example:
+
+- All `String` values (both `StringScalars` and `StringColumns`) have the
+  method `.upper()` that transforms the string to uppercase.
+  `Floating` and `Array` values don't have this method, of course.
+- `IntegerColumn` and `FloatingColumn`  values have `.mean()`, `.max()`, etc methods
+  (which return `FloatingScalar` values),
+  because you can aggregate over them, since they are a collection of values.
+  On the other hand, `IntegerScalar` and `FloatingColumn` values do **not** have these
+  methods, because it doesn't make sense to take the mean or max of a single value.
+- If you call `.execute()` on these values, you get different results.
```

Comment:
```suggestion
- If you call `.to_pandas()` on these values, you get different results.
```

or `to_pyarrow`?  but we're trying to get away from `execute`

Reply:
ahh, I see below you are referencing pandas datashapes, so definitely `to_pandas`, then